lib/karto_sdk/src/Mapper.cpp

```diff
@@ -2548,25 +2601,38 @@ void Mapper::setParamUseResponseExpansion(bool b)
 
 void Mapper::Initialize(kt_double rangeThreshold)
 {
-  if (m_Initialized == false) {
-    // create sequential scan and loop matcher
-    m_pSequentialScanMatcher = ScanMatcher::Create(this,
-        m_pCorrelationSearchSpaceDimension->GetValue(),
-        m_pCorrelationSearchSpaceResolution->GetValue(),
-        m_pCorrelationSearchSpaceSmearDeviation->GetValue(),
-        rangeThreshold);
-    assert(m_pSequentialScanMatcher);
+  if (m_Initialized) {
+    return;
+  }
+  // create sequential scan and loop matcher, update if deserialized
 
+  if (m_pSequentialScanMatcher) {
+    delete m_pSequentialScanMatcher;
+  }
+  m_pSequentialScanMatcher = ScanMatcher::Create(this,
+      m_pCorrelationSearchSpaceDimension->GetValue(),
+      m_pCorrelationSearchSpaceResolution->GetValue(),
+      m_pCorrelationSearchSpaceSmearDeviation->GetValue(),
+      rangeThreshold);
+  assert(m_pSequentialScanMatcher);
+
+  if (m_Deserialized) {
+    m_pMapperSensorManager->SetRunningScanBufferSize(m_pScanBufferSize->GetValue());
+    m_pMapperSensorManager->SetRunningScanBufferMaximumDistance(m_pScanBufferMaximumScanDistance->GetValue());
+
+    m_pGraph->UpdateLoopScanMatcher(rangeThreshold);
+  }
+  else {
     m_pMapperSensorManager = new MapperSensorManager(m_pScanBufferSize->GetValue(),
```

Comment:
Why not just create the new object of this either way if its getting the new parameter value?

Reply:
because `MapperSensorManager` holds the `ScanMangers` with the respective scans. I think that is something we want to keep.

Reply:
Oh sorry I thought I resolved this - dont worry about it that was my mistake I was deleting